internal/rbd/nodeserver.go

```diff
@@ -140,6 +143,72 @@ func healerStageTransaction(ctx context.Context, cr *util.Credentials, volOps *r
 	return nil
 }
 
+// parseMapOptions helps parse formatted mapOptions and unmapOptions and
+// returns mounter specific options.
+func parseMapOptions(mapOptions string) (string, string, error) {
+	var krbdMapOptions, nbdMapOptions string
+	const (
+		noKeyLength = 1
+		validLength = 2
+	)
+	for _, item := range strings.Split(mapOptions, ";") {
+		var mounter, options string
+		if item == "" {
+			continue
+		}
+		s := strings.Split(item, ":")
+		switch len(s) {
+		case noKeyLength:
+			options = strings.TrimSpace(s[0])
+			krbdMapOptions = options
+		case validLength:
+			mounter = strings.TrimSpace(s[0])
+			options = strings.TrimSpace(s[1])
+			switch strings.ToLower(mounter) {
+			case "krbd":
+				krbdMapOptions = options
+			case "nbd":
+				nbdMapOptions = options
+			default:
+				return "", "", errMounterNotFound
+			}
+		default:
+			return "", "", errBadlyFormattedOpts
+		}
+	}
+
+	return krbdMapOptions, nbdMapOptions, nil
+}
+
+// getMapOptions is a wrapper func, calls parse map/unmap funcs and feeds the
+// rbdVolume object.
+func getMapOptions(req *csi.NodeStageVolumeRequest, rv *rbdVolume) error {
+	var (
+		krbdMapOptions   string
+		nbdMapOptions    string
+		krbdUnmapOptions string
+		nbdUnmapOptions  string
+	)
+
+	krbdMapOptions, nbdMapOptions, err := parseMapOptions(req.GetVolumeContext()["mapOptions"])
+	if err != nil {
+		return err
+	}
+	krbdUnmapOptions, nbdUnmapOptions, err = parseMapOptions(req.GetVolumeContext()["unmapOptions"])
+	if err != nil {
+		return err
+	}
+	if rv.Mounter == rbdDefaultMounter {
+		rv.MapOptions = krbdMapOptions
+		rv.UnmapOptions = krbdUnmapOptions
+	} else if rv.Mounter == rbdNbdMounter {
+		rv.MapOptions = nbdMapOptions
+		rv.UnmapOptions = nbdUnmapOptions
+	}
```

Comment:
```suggestion
	krbdMapOptions, nbdMapOptions, err := parseMapOptions(req.GetVolumeContext()["mapOptions"])
	if err != nil {
		return err
	}
	krbdUnmapOptions, nbdUnmapOptions, err := parseMapOptions(req.GetVolumeContext()["unmapOptions"])
	if err != nil {
		return err
	}
	if rv.Mounter == rbdDefaultMounter {
		rv.MapOptions = krbdMapOptions
		rv.UnmapOptions = krbdUnmapOptions
	} else if rv.Mounter == rbdNbdMounter {
		rv.MapOptions = nbdMapOptions
		rv.UnmapOptions = nbdUnmapOptions
	}
```

variables declarations can be avoided?